### ISSUE TITLE
Add access to originally thrown error in `createJsonQuery` and other factories

### ISSUE THREAD

**Issue by AlexandrHoroshih** [igorkamyshev/farfetched#283]
Hello!

## Context

There are cases, when `farfetched` queries provide their own custom errors like 
```json
{
   "error":{
      "errorType":"NETWORK",
      "explanation":"Request was failed due to network problems",
      "reason":"Failed to fetch"
   },
   "meta":{
      "stopErrorPropagation":false,
      "isFreshData":true
   }
}
```
and, while it is super cool most of the times, in these cases there is no access to original error, which leads to a problem - Some other meta information, that may be presentend on original error object, is lost, [because internally `farfetched` only takes a `message` field from original object ](https://github.com/igorkamyshev/farfetched/blob/cd0820a5d39e811ac94adbbbcf026459fec728fc/packages/core/src/fetch/request.ts#L19)

## Suggestion

To fix this issue, i suggest to always provide original error object as a special field like `originalError` or `catched`

```json
{
   "error":{
      "errorType":"NETWORK",
      "explanation":"Request was failed due to network problems",
      "reason":"Failed to fetch"
   },
   "catched": ... // original thing, that was catched in `catch`
   "meta":{
      "stopErrorPropagation":false,
      "isFreshData":true
   }
}
```

**Comment by igorkamyshev** [comment]
Will be done in 0.8

**Comment by igorkamyshev** [comment]
Done in https://github.com/igorkamyshev/farfetched/commit/386752e2ca465ced12ad0d4cd1c2a5f417126e8d, will be released in 0.8.
